src/pages/settings/ProfilePage.js

```diff
@@ -1,24 +1,243 @@
-import React from 'react';
+import React, {Component} from 'react';
+import {withOnyx} from 'react-native-onyx';
+import PropTypes from 'prop-types';
+import {
+    View,
+    TextInput,
+} from 'react-native';
+import RNPickerSelect from 'react-native-picker-select';
+import Str from 'expensify-common/lib/str';
+import moment from 'moment-timezone';
+
 import HeaderWithCloseButton from '../../components/HeaderWithCloseButton';
 import Navigation from '../../libs/Navigation/Navigation';
 import ScreenWrapper from '../../components/ScreenWrapper';
+import {setPersonalDetails} from '../../libs/actions/PersonalDetails';
 import ROUTES from '../../ROUTES';
+import ONYXKEYS from '../../ONYXKEYS';
+import CONST from '../../CONST';
+import Avatar from '../../components/Avatar';
+import styles from '../../styles/styles';
+import Text from '../../components/Text';
+import {DownArrow} from '../../components/Icon/Expensicons';
+import Icon from '../../components/Icon';
+import Checkbox from '../../components/Checkbox';
+import ButtonWithLoader from '../../components/ButtonWithLoader';
+
+const propTypes = {
+    /* Onyx Props */
+    // The personal details of the person who is logged in
+    myPersonalDetails: PropTypes.shape({
+        // Email/Phone login of the current user from their personal details
+        login: PropTypes.string,
+
+        // Display first name of the current user from their personal details
+        firstName: PropTypes.string,
+
+        // Display last name of the current user from their personal details
+        lastName: PropTypes.string,
+
+        // Avatar URL of the current user from their personal details
+        avatarURL: PropTypes.string,
+
+        // Pronouns of the current user from their personal details
+        pronouns: PropTypes.string,
+
+        // timezone of the current user from their personal details
+        timezone: PropTypes.shape({
+
+            // Value of selected timezone
+            selected: PropTypes.string,
+
+            // Whether timezone is automatically set
+            automatic: PropTypes.bool,
+        }),
+    }),
+};
+
+const defaultProps = {
+    myPersonalDetails: {},
+};
+
+const timezones = moment.tz.names()
+    .map(timezone => ({
+        value: timezone,
+        label: timezone,
+    }));
+
+class ProfilePage extends Component {
+    constructor(props) {
+        super(props);
+
+        const {
+            firstName,
+            lastName,
+            pronouns,
+            timezone = {},
+        } = props.myPersonalDetails;
+        const pronounValues = Object.values(CONST.PRONOUNS);
+
+        let initialPronouns = pronouns;
+        let initialSelfSelectedPronouns = '';
+
+        if (pronouns && !pronounValues.includes(pronouns)) {
```

Comment:
Might be good to add a comment here explaining that you're handling self-selected pronouns here.